src/rgw/rgw_bucket.cc

```diff
@@ -295,29 +295,28 @@ int rgw_bucket_instance_remove_entry(RGWRados *store, string& entry, RGWObjVersi
   return store->meta_mgr->remove_entry(bucket_instance_meta_handler, entry, objv_tracker);
 }
 
+// parse instance in format: [tenant:]instance[:shard_id], returning
```

Comment:
is it safe to assume that all bucket instances will have the format "name:zone.instance_id.bucket_id" that results from this line in `RGWRados::create_bucket()`?
```c++
snprintf(buf, sizeof(buf), "%s.%llu.%llu", get_zone_params().get_id().c_str(), (long long)iid, (long long)bid);
```
if so, we can resolve the ambiguity by parsing ':' as shard_id only if it comes after the '.'s